membership-attribute-service/app/services/ZuoraAttributeService.scala

```diff
@@ -0,0 +1,141 @@
+package services
+import com.gu.memsub.Subscription.AccountId
+import com.gu.memsub.subsv2.Subscription
+import com.gu.memsub.subsv2.SubscriptionPlan.AnyPlan
+import com.gu.memsub.subsv2.reads.SubPlanReads
+import com.gu.memsub.subsv2.reads.SubPlanReads.anyPlanReads
+import com.gu.zuora.ZuoraRestService.QueryResponse
+import loghandling.LoggingField.LogField
+import loghandling.{LoggingWithLogstashFields, ZuoraRequestCounter}
+import models.Attributes
+import org.joda.time.LocalDate
+import play.api.libs.concurrent.Execution.Implicits._
+
+import scala.concurrent.Future
+import scalaz.std.list._
+import scalaz.std.scalaFuture._
+import scalaz.syntax.traverse._
+import scalaz.{-\/, Disjunction, EitherT, \/, \/-, _}
+
+
+class ZuoraAttributeService(identityIdToAccountIds: String => Future[String \/ QueryResponse],
+                            subscriptionsForAccountId: AccountId => SubPlanReads[AnyPlan] => Future[Disjunction[String, List[Subscription[AnyPlan]]]],
+                            dynamoAttributeGetter: String => Future[Option[Attributes]]) extends LoggingWithLogstashFields {
+
+  def get(userId: String): Future[Option[Attributes]] = attributesFromZuora(userId)
```

Comment:
get and attributesFromZuora are the same function, no need to have both?

Reply:
regarding naming, since (from a dependencies point of view) we want to keep this code only doing the one thing, it might be worth removing the word "service" and calling it AttributesFromZuora.  Then people won't be tempted to add any other functions in the same file.
Ideally we could make this into an object instead of a class and avoid adding more to touchpointbackend, but that would need more refactoring.

Reply:
re: get and attributesFromZuora
Agreed. I didn't mean to leave get behind so thanks for spotting that. 

re: an object AttributesFromZuora
I like this suggestion and I've tried to take it. Can you take another look?